frame/staking/src/benchmarking.rs

```diff
@@ -477,6 +405,196 @@ benchmarks! {
+
+		// set number of winners
+		ValidatorCount::put(w);
+
+		// create a assignments in total for the w winners.
+		let (winners, assignments) = create_assignments_for_offchain::<T>(a, winners)?;
+
+		let single_winner = winners[0].0.clone();
+
+		let (
+			winners,
+			compact,
+			score,
+			size
+		) = offchain_election::prepare_submission::<T>(assignments, winners, false).unwrap();
+
+		// needed for the solution to be accepted
+		<EraElectionStatus<T>>::put(ElectionStatus::Open(T::BlockNumber::from(1u32)));
+
+		let era = <Staking<T>>::current_era().unwrap_or(0);
+		let caller: T::AccountId = account("caller", n, SEED);
+
+		// submit a very bad solution on-chain
+		{
+			// this is needed to fool the chain to accept this solution.
+			ValidatorCount::put(1);
+			let (winners, compact, score, size) = get_single_winner_solution::<T>(single_winner)?;
+			assert!(
+				<Staking<T>>::submit_election_solution(
+					RawOrigin::Signed(caller.clone()).into(),
+					winners,
+					compact,
+					score.clone(),
+					era,
+					size,
+			).is_ok());
+
+			// new solution has been accepted.
+			assert_eq!(<Staking<T>>::queued_score().unwrap(), score);
+			ValidatorCount::put(w);
+		}
+	}: {
+		let result = <Staking<T>>::submit_election_solution(
+			RawOrigin::Signed(caller.clone()).into(),
+			winners,
+			compact,
+			score.clone(),
+			era,
+			size,
+		);
+		assert!(result.is_ok());
+	}
+	verify {
+		// new solution has been accepted.
+		assert_eq!(<Staking<T>>::queued_score().unwrap(), score);
+	}
+
+	// This will be early rejected based on the score.
+	submit_solution_weaker {
+		// number of validator intent
+		let v in 1000 .. 2000;
+		// number of nominator intent
+		let n in 1000 .. 2000;
+
+		MinimumValidatorCount::put(0);
+		create_validators_with_nominators_for_era::<T>(v, n, MAX_NOMINATIONS, false, None)?;
+
+		// needed for the solution to be generates.
+		assert!(<Staking<T>>::create_stakers_snapshot().0);
+
+		// needed for the solution to be accepted
+		<EraElectionStatus<T>>::put(ElectionStatus::Open(T::BlockNumber::from(1u32)));
+		let caller: T::AccountId = account("caller", n, SEED);
+		let era = <Staking<T>>::current_era().unwrap_or(0);
+
+		// submit a seq-phragmen with all the good stuff on chain
+		{
+			let (winners, compact, score, size) = get_seq_phragmen_solution::<T>(true);
+			assert!(
+				<Staking<T>>::submit_election_solution(
+					RawOrigin::Signed(caller.clone()).into(),
+					winners,
+					compact,
+					score.clone(),
+					era,
+					size,
+			).is_ok());
+
+			// new solution has been accepted.
+			assert_eq!(<Staking<T>>::queued_score().unwrap(), score);
+		}
+
+		// prepare a bad solution. This will be very early rejected.
+		let (winners, compact, score, size) = get_weak_solution::<T>(true);
+	}: {
+		assert!(
+			<Staking<T>>::submit_election_solution(
+				RawOrigin::Signed(caller.clone()).into(),
+				winners,
+				compact,
+				score.clone(),
+				era,
+				size,
+		).is_err());
```

Comment:
Maybe better to assert the kind of err it is.

Reply:
assert_err is not available here and doing it manually, I worry it might influence the bench too much.

Reply:
yes ok